guardowl/protocols.py

```diff
@@ -382,17 +159,12 @@ def _run_simulation(
             """
         )
 
-        system = parameters.system
-
-        qsim = SimulationFactory.create_simulation(
-            system,
-            parameters.testsystem.topology,
-            platform=parameters.platform,
-            temperature=temperature,
-            env=parameters.env,
-            device_index=parameters.device_index,
-            ensemble=ensemble,
-        )
+    @staticmethod
+    def _run_simulation(
+        parameters: StabilityTestParameters,
+        qsim: Simulation,
+    ) -> None:
+        from openmm.app import DCDReporter, PDBFile
 
         os.makedirs(parameters.output_folder, exist_ok=True)
```

Comment:
Not sure what your thoughts are on using os vs. pathlib for these kinds of operations. Pathlib is the modern way to do such things but personally I am not massively bothered but just thought to mention.

Reply:
Ah, absolutly. Will switch to pathlib.